### PR TITLE
pull up c-band prefix decoding

### PR DESCRIPTION
**Pull request by rpatel3001**
doing it this way means using top level MessageDecoder instance in all the tests instead of an instance of the specific plugin being tested. if that's ok i can update the rest to match (and check if the C-Band prefix test matches any messages it shouldn't).

### CODE REVIEW

File: lib/plugins/Label_4A.test.ts [hunk 1]
```diff
@@ -42,19 +41,18 @@ test('decodes Label 4A, variant 1', () => {
 
 test('decodes Label 4A, variant 1, no callsign', () => {
   const decoder = new MessageDecoder();
-  const decoderPlugin = new Label_4A(decoder);
 
   // https://app.airframes.io/messages/3452310240
   const text = '101606,1910,.N317FR,,KMDW,----,1,1016,RT0,LT1,';
-  const decodeResult = decoderPlugin.decode({ text: text });
+  const decodeResult = decoder.decode({ label: "4A", text: text });
```
Comment (makrsmark) [inline lib/plugins/Label_4A.test.ts:47]: this isn't a bad idea, but it's going to make the PR huge. Do we want a separate PR that just makes this change? Or do we want to move the c-band messages to the MessageDecoder suite? I'm leaning towards moving the c-band tests

Comment (rpatel3001) [reply lib/plugins/Label_4A.test.ts:47]: Moving the c band message tests makes sense with two drawbacks:

1. Need to remember to add tests in a second location for types with this prefix
2. More importantly, there's no way to check if the cband prefix accidentally catches non cband messages and incorrectly truncates the message that it passes into a plugin.

Comment (makrsmark) [reply lib/plugins/Label_4A.test.ts:47]: do we have messages (minus the prefix) that are unique to c-band? If not, i don't think we need to have c-band label-specific tests other than a couple to test the prefix extraction. 

As for the second point, idk. Is it worth slowing down the test suite by making every test run through the plugin selection code just so we can verify this edge case that we don't even know exists?

Comment (rpatel3001) [reply lib/plugins/Label_4A.test.ts:47]: not sure about any cband specific messages, but the regex for the prefix is pretty permissive. i would be surprised if it never matched any non cband messages. maybe it would be better to send more metadata about the source of the message into the decoders and just tell it explicitly if it's cband?

or we could try to decode as cband if the prefix is detected then retry with the prefix still attached if it fails to decode. that wouldn't necessarily catch every edge case though

Comment (makrsmark) [reply lib/plugins/Label_4A.test.ts:47]: @kevinelliott and @andermatt64 - Do you have any opinions here?

File: lib/MessageDecoder.ts [hunk 2]
```diff
@@ -87,6 +88,14 @@ export class MessageDecoder {
       }
     }
 
+    // C-Band puts a 10 char header in front of some message types
+    // First 4 chars are some kind of message number
+    // Last 6 chars are the flight number
+    let cband = message.text.match(/^(?<msgno>[A-Z]\d{2}[A-Z])(?<airline>[A-Z]{2})(?<number>[0-9]{4})/);
```
Comment (andermatt64) [inline lib/MessageDecoder.ts:94]: For `airline` capture group, we might have to be a bit more forgiving in terms of what is allowed since the following are valid IATA callsigns: `EK05RQ`, `B60001`, `9E0001`, `MCEY42`

Comment (rpatel3001) [reply lib/MessageDecoder.ts:94]: seems like there's a possible full format described here: https://en.wikipedia.org/wiki/Airline_codes#IATA_airline_designator

It makes sense to let the airline be letters or digits, but the C-Band prefix is a fixed 10 characters as far as I've seen, and it pads out the numeric part to 4 digits, so there's no room for the third airline designator character or optional operational suffix. I could change it anyway if you like but I think it makes sense to try and keep it as tight as possible to try to avoid false matches.

**Comment by kevinelliott** [comment]
@rpatel3001 @andermatt64 @makrsmark Any further thoughts here? It would be good to get this one finished up.

**Comment by andermatt64** [comment]
> @rpatel3001 @andermatt64 @makrsmark Any further thoughts here? It would be good to get this one finished up.

I think I'm ok with the airline+number regex if historical C-band messages can be consistently caught with it.

**Comment by rpatel3001** [comment]
last thing to resolve is the conversation about how to do c-band tests: make each plugin test use `decoder` instead of `decoderPlugin` or do a handful of c-band tests as part of the MessageDecoder test suite.

**Comment by kevinelliott** [comment]
@coderabbitai full review

**Comment by rpatel3001** [comment]
revisiting this, I think I agree that adding c-band tests to each message type is perhaps too much and so it would be better to have a c-band specific test file. i can pull them out and revert the existing tests in the next couple days

File: lib/plugins/CBand.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          text: messageText,
+        }, options);
+        if(decoded.decoded) {
+          ResultFormatter.flightNumber(decodeResult, cband.groups.airline + Number(cband.groups.number));
+          decodeResult.decoded = true;
+          decodeResult.decoder.decodeLevel = decoded.decoder.decodeLevel;
+          decodeResult.decoder.name = this.name + '-' + decoded.decoder.name;
+          decodeResult.raw = {...decodeResult.raw, ...decoded.raw };
+          decodeResult.formatted.description = decoded.formatted.description;
+          decodeResult.formatted.items.push(...decoded.formatted.items);
```
[the remark this answers is not in the document]
Comment (makrsmark) [reply lib/plugins/CBand.ts]: Good question - do we want `XA1` or `XA0001` ?
